### PR TITLE
playermanager: add.

### LINKED ISSUE #1942: prayagainstplayer: NPE
```java
java.lang.NullPointerException: null
    at net.runelite.client.util.ImageUtil.resizeCanvas(ImageUtil.java:213)
    at net.runelite.client.plugins.prayagainstplayer.PrayAgainstPlayerPlugin.ProtectionIconFromSprite(PrayAgainstPlayerPlugin.java:410)
    at net.runelite.client.plugins.prayagainstplayer.PrayAgainstPlayerPlugin.loadProtectionIcons(PrayAgainstPlayerPlugin.java:378)
    at net.runelite.client.plugins.prayagainstplayer.PrayAgainstPlayerPlugin.onGameStateChanged(PrayAgainstPlayerPlugin.java:150)
    at io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:63)
    at io.reactivex.internal.operators.observable.ObservableMap$MapObserver.onNext(ObservableMap.java:62)
    at io.reactivex.internal.operators.observable.ObservableFilter$FilterObserver.onNext(ObservableFilter.java:52)
    at com.jakewharton.rxrelay2.PublishRelay$PublishDisposable.onNext(PublishRelay.java:180)
    at com.jakewharton.rxrelay2.PublishRelay.accept(PublishRelay.java:145)
    at com.jakewharton.rxrelay2.SerializedRelay.accept(SerializedRelay.java:57)
    at net.runelite.client.eventbus.EventBus.post(EventBus.java:102)
    at net.runelite.client.callback.Hooks.post(Hooks.java:161)
    at client.gameStateChanged(client.java:1112)
    at dh.fp(dh.java:1752)
    at client.ae(client.java:1594)
    at bg.av(bg.java:355)
    at bg.run(bg.java:334)
    at java.lang.Thread.run

```

### LINKED ISSUE #1950: playerscouter: NoSuchElementException
```java
java.util.NoSuchElementException: null
    at java.util.LinkedHashMap$LinkedHashIterator.nextNode
    at java.util.LinkedHashMap$LinkedEntryIterator.next
    at java.util.LinkedHashMap$LinkedEntryIterator.next
    at net.runelite.client.plugins.playerscouter.PlayerScouter.getEntry(PlayerScouter.java:448)
    at net.runelite.client.plugins.playerscouter.PlayerScouter.scoutPlayer(PlayerScouter.java:579)
    at net.runelite.client.plugins.playerscouter.PlayerScouter.lambda$onGameTick$0(PlayerScouter.java:201)
    at java.lang.Iterable.forEach
    at net.runelite.client.plugins.playerscouter.PlayerScouter.onGameTick(PlayerScouter.java:190)
    at io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:63)
    at io.reactivex.internal.operators.observable.ObservableMap$MapObserver.onNext(ObservableMap.java:62)
    at io.reactivex.internal.operators.observable.ObservableFilter$FilterObserver.onNext(ObservableFilter.java:52)
    at com.jakewharton.rxrelay2.PublishRelay$PublishDisposable.onNext(PublishRelay.java:180)
    at com.jakewharton.rxrelay2.PublishRelay.accept(PublishRelay.java:145)
    at com.jakewharton.rxrelay2.SerializedRelay.accept(SerializedRelay.java:57)
    at net.runelite.client.eventbus.EventBus.post(EventBus.java:102)
    at net.runelite.client.callback.Hooks.clientMainLoop(Hooks.java:179)
    at client.draw(client.java:1490)
    at client.au(client.java)
    at bg.am(bg.java:381)
    at bg.run(bg.java:335)
    at java.lang.Thread.run
```

### LINKED ISSUE #1972: playerscouter: NPE
```java
java.lang.NullPointerException: null
    at java.util.Objects.requireNonNull
    at net.runelite.client.plugins.playerscouter.PlayerScouter.location(PlayerScouter.java:646)
    at net.runelite.client.plugins.playerscouter.PlayerScouter.update(PlayerScouter.java:291)
    at net.runelite.client.plugins.playerscouter.PlayerScouter.lambda$onGameTick$0(PlayerScouter.java:192)
    at java.lang.Iterable.forEach
    at net.runelite.client.plugins.playerscouter.PlayerScouter.onGameTick(PlayerScouter.java:190)
    at io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:63)
    at io.reactivex.internal.operators.observable.ObservableMap$MapObserver.onNext(ObservableMap.java:62)
    at io.reactivex.internal.operators.observable.ObservableFilter$FilterObserver.onNext(ObservableFilter.java:52)
    at com.jakewharton.rxrelay2.PublishRelay$PublishDisposable.onNext(PublishRelay.java:180)
    at com.jakewharton.rxrelay2.PublishRelay.accept(PublishRelay.java:145)
    at com.jakewharton.rxrelay2.SerializedRelay.accept(SerializedRelay.java:57)
    at net.runelite.client.eventbus.EventBus.post(EventBus.java:102)
    at net.runelite.client.callback.Hooks.clientMainLoop(Hooks.java:179)
    at client.draw(client.java:1490)
    at client.au(client.java)
    at bg.am(bg.java:381)
    at bg.run(bg.java:335)
    at java.lang.Thread.run
```

### PR DESCRIPTION
**Pull request by Ganom**
playerscouter: update to use new player manager.
playerscouter: add minify option.

Closes #1972 
Closes #1950
Closes #1942

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/game/PlayerManager.java [hunk 1]
```diff
@@ -0,0 +1,504 @@
...
+
+			resultCache.put(player.getName(), result);
+			player.setSkills(result);
+			player.setPrayerLevel(player.getSkills().getPrayer().getLevel());
+			player.setHpLevel(player.getSkills().getHitpoints().getLevel());
+			player.setHttpRetry(false);
+		});
+	}
+
+	private void onAppearenceChanged(PlayerAppearanceChanged event)
```
Comment (stone-wall) [inline runelite-client/src/main/java/net/runelite/client/game/PlayerManager.java]:
```suggestion
	private void onAppearenceChanged(PlayerAppearanceChanged event)
	{
		if (event.getPlayer() == null)
		{
			return;
		}
```

File: runelite-client/src/main/java/net/runelite/client/game/PlayerManager.java [hunk 2]
```diff
@@ -0,0 +1,504 @@
...
+	{
+		final PlayerContainer player = playerMap.get(event.getPlayer().getName());
+
+		if (player == null)
```
Comment (stone-wall) [inline runelite-client/src/main/java/net/runelite/client/game/PlayerManager.java]:
```suggestion
```

File: runelite-client/src/main/java/net/runelite/client/game/PlayerManager.java [hunk 3]
```diff
@@ -0,0 +1,504 @@
...
+		{
```
Comment (stone-wall) [inline runelite-client/src/main/java/net/runelite/client/game/PlayerManager.java]:
```suggestion
```

File: runelite-client/src/main/java/net/runelite/client/game/PlayerManager.java [hunk 4]
```diff
@@ -0,0 +1,504 @@
...
+			return;
```
Comment (stone-wall) [inline runelite-client/src/main/java/net/runelite/client/game/PlayerManager.java]:
```suggestion
```

File: runelite-client/src/main/java/net/runelite/client/game/PlayerManager.java [hunk 5]
```diff
@@ -0,0 +1,504 @@
...
+		}
```
Comment (stone-wall) [inline runelite-client/src/main/java/net/runelite/client/game/PlayerManager.java]:
```suggestion
```

File: runelite-client/src/main/java/net/runelite/client/game/PlayerManager.java [hunk 6]
```diff
@@ -0,0 +1,504 @@
...
+			resultCache.put(player.getName(), result);
+			player.setSkills(result);
+			player.setPrayerLevel(player.getSkills().getPrayer().getLevel());
+			player.setHpLevel(player.getSkills().getHitpoints().getLevel());
+			player.setHttpRetry(false);
+		});
+	}
+
+	private void onAppearenceChanged(PlayerAppearanceChanged event)
+	{
```
Comment (stone-wall) [inline runelite-client/src/main/java/net/runelite/client/game/PlayerManager.java]:
```suggestion
```

File: runelite-client/src/main/java/net/runelite/client/game/PlayerManager.java [hunk 7]
```diff
@@ -0,0 +1,504 @@
...
+		}
+
+		update(player);
+		player.reset();
+		player.setFriend(client.isFriended(player.getName(), false));
+		player.setClan(client.isClanMember(player.getName()));
+	}
+
+	private void onPlayerDespawned(PlayerDespawned event)
+	{
```
Comment (stone-wall) [inline runelite-client/src/main/java/net/runelite/client/game/PlayerManager.java]:
```suggestion
	{
	if (event.getPlayer() == null)
	{
		return;
	}
```

File: runelite-client/src/main/java/net/runelite/client/game/PlayerManager.java [hunk 8]
```diff
@@ -0,0 +1,504 @@
...
+		final Player player = event.getPlayer();
+		playerMap.remove(player.getName());
+	}
+
+	private void onPlayerSpawned(PlayerSpawned event)
+	{
```
Comment (stone-wall) [inline runelite-client/src/main/java/net/runelite/client/game/PlayerManager.java]:
```suggestion
	{
	if (event.getPlayer() == null)
	{
		return;
	}
```

File: runelite-client/src/main/java/net/runelite/client/game/PlayerManager.java [hunk 9]
```diff
@@ -0,0 +1,504 @@
...
+		final Player player = event.getPlayer();
+		playerMap.put(player.getName(), new PlayerContainer(player));
+	}
+
+	private void onAnimationChanged(AnimationChanged event)
+	{
```
Comment (stone-wall) [inline runelite-client/src/main/java/net/runelite/client/game/PlayerManager.java]:
```suggestion
	{
	if (event.getActor() == null)
	{
		return;
	}
```

File: runelite-client/src/main/java/net/runelite/client/game/PlayerContainer.java [hunk 10]
```diff
@@ -0,0 +1,191 @@
...
+	private int rangeAttack;
+	private int rangeDefence;
+	private int rangeStr;
+	private int risk;
+	private int scoutTimer;
+	private int shield;
+	private int speed;
+	private int timer;
+	private int weapon;
+	private int wildyLevel;
```
[the remark this answers is not in the document]
Comment (Lucwousin) [reply runelite-client/src/main/java/net/runelite/client/game/PlayerContainer.java]: wait I said that badly. Divided up into newer value objects, with for instance combat stats in 1 object, gear stats in 1, etc

**Pull request merged**